src/routes/user/deleteUser.ts

```diff
@@ -0,0 +1,61 @@
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { Type } from '@sinclair/typebox';
+import { knexController } from '../../database/database';
+import {
+  userIdParamsPayload,
+  UserIdParamsPayload,
+} from '../../types/user.types';
+
+const deleteUserReply = Type.Object({
+  userId: Type.String(),
+  deletedAt: Type.String(),
+});
+
+const schema = {
+  description: 'Delete (=anonymize) user with given userId',
+  summary: 'Delete user',
+  tags: ['User'],
+  params: userIdParamsPayload,
+  response: {
+    200: deleteUserReply,
+    401: { $ref: 'error' },
+    403: { $ref: 'error' },
+    404: { $ref: 'error' },
+  },
+};
+
+const handler = async (
+  req: FastifyRequest<{ Params: UserIdParamsPayload }>,
+  reply: FastifyReply
+): Promise<void> => {
+  // TODO: Authorization
+  const userId: string = req.params.userId;
+  const result = await knexController('user').where({ id: userId }).update({
+    email: null,
+    forename: null,
+    surname: null,
+    is_admin: false,
+    is_blender: false,
```

Comment:
Käyttäjän poistossa pitää säästää käyttäjän täyttöhistoria laskutusta varten, toiminnantarkastajaa varten, yms. Ei siis liene tarpeellista anonymisoida näitä kenttiä? Vai onko? is_admin saattaa olla yksilöivä tieto, mutta is_blender ei varmastikkaan.

Tää voi olla hyvä näin tälläikää, mutta jättäisin vaikka issuen tästä, ettei unohdeta miettiä tätä myöhemmin.

Reply:
Toisaalta laskutuksen kannalta käyttäjän roolit ei ole ole olennaisia. Mutta tee vaan tiketti